lib/drivers.js

```diff
@@ -0,0 +1,157 @@
+      msg += `Drivers.json entry for '${driverName}' has missing ` +
+             `or incorrect version. `;
+    }
+
+    if (!_.isString(INSTALLED_DRIVERS[driverName].pkg)) {
+      msg += `Drivers.json entry for '${driverName}' has missing ` +
+             `or incorrect NPM package name. `;
+    }
+
+    if (!_.isString(INSTALLED_DRIVERS[driverName].mainClass)) {
+      msg += `Drivers.json entry for '${driverName}' has missing ` +
+             `or incorrect class name. `;
+    }
+  }
+
+  if (msg) {
+    throw new Error(`Appium encountered an error while trying to verify ` +
+              `the data contained in the drivers file (${DRIVERS_JSON}): ` +
+              `${msg}Please ensure you have installed drivers correctly ` +
+              `using the Appium CLI.`);
+  }
+
+  return INSTALLED_DRIVERS;
+}
+
+function printAvailableDrivers () {
+  const driverNames = Object.keys(INSTALLED_DRIVERS);
+  if (!driverNames.length) {
+    log.info('No drivers have been installed. Use the "appium driver" ' +
+             'command to install the one(s) you want to use.');
+    return;
+  }
+
+  log.info('Available drivers:');
+  for (const driverName of Object.keys(INSTALLED_DRIVERS)) {
+    const ver = INSTALLED_DRIVERS[driverName].version;
+    const aName = INSTALLED_DRIVERS[driverName].automationName;
+    log.info(`  - ${driverName}@${ver} (automationName '${aName}')`);
+  }
+}
+
+function getDriverBySupport (automationName, platformName) {
+  for (const driverName of Object.keys(INSTALLED_DRIVERS)) {
+    const installed = INSTALLED_DRIVERS[driverName];
+    const aNameMatches = installed.automationName.toLowerCase() ===
+                         automationName.toLowerCase();
+    const lower = (x) => x.toLowerCase();
+    const pNameMatches = _.includes(installed.platformNames.map(lower),
+                                    platformName.toLowerCase());
+
+    if (aNameMatches && pNameMatches) {
+      return {driverName, ...installed};
+    } else if (aNameMatches) {
+      throw new Error(`Driver '${driverName}' supports automationName ` +
+                      `'${automationName}', but Appium could not find ` +
+                      `support for platformName '${platformName}'. Supported ` +
+                      `platformNames are: ` +
+                      JSON.stringify(installed.platformNames));
+    }
+  }
+
+  throw new Error(`Could not find installed driver to support given caps`);
+}
+
+function findMatchingDriver ({automationName, platformName}) {
+  if (!_.isString(platformName)) {
+    throw new Error('You must include a platformName capability');
+  }
+
+  if (!_.isString(automationName)) {
+    throw new Error('You must include an automationName capability');
+  }
+
+  log.info(`Attempting to find matching driver for automationName ` +
+           `'${automationName}' and platformName '${platformName}'`);
+
+  try {
+    const {driverName, mainClass, pkg, version} = getDriverBySupport(automationName, platformName);
```

Comment:
what the function name means (BySupport) ?

Reply:
get the driver based on its supported automation name and platform names. kinda long-winded :-)